File: code/library/src/main/java/org/fbme/lib/iec61499/declarations/ApplicationDeclaration.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Declaration;
import org.fbme.lib.iec61499.fbnetwork.FBNetwork;
import org.fbme.lib.iec61499.fbnetwork.subapp.SubappNetwork;
import org.jetbrains.annotations.NotNull;

public interface ApplicationDeclaration extends Declaration, ContainedElement {

    @NotNull SubappNetwork getNetwork();

    @NotNull SystemDeclaration getContainer();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/ApplicationDeclaration.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Declaration
import org.fbme.lib.iec61499.fbnetwork.subapp.SubappNetwork

interface ApplicationDeclaration : DeclarationWithNetwork, ContainedElement {
    override val network: SubappNetwork
    override val container: SystemDeclaration
}
